Marshal branch

MarshMonci opened the pull request:
Made request for SOD, PSSOD, and Role Separation

MarshMonci committed:
Add files via upload

Added personal Role files

MarshMonci committed:
Add files via upload

Added the separation for SOD and PSSOD

MarshMonci committed:
Delete Role.cpp

MarshMonci committed:
Add files via upload

Updated to temporarily revert changes to Role.h as I try to find a solution to errors compiling with separation. Also updated SOD to include functional separation.

MarshMonci committed:
Add files via upload

Added functional separation for UserRoleRelation and UserRoleAssignment.

Iceylands reviewed (changes requested):
Review and update the solution, if you're adding or deleting files the new, updated solution has to be uploaded as well.

Iceylands committed:
Merge branch 'master' into Marshal-branch

Iceylands commented:
I solved the conflicts between the two UserRoleRelation files and merged them, just need the solution file fixed.

MarshMonci committed:
Add files via upload

Updated solution file to hopefully resolve issues with UserRoleRelation & UserRoleActivation. Also created WIP SystemManager class (separated). The driver has been slightly modified to be in line with SystemManager functionality, though this also needs to be finished. As far as I can tell there are no issues with this version and it does run correctly.

MarshMonci committed:
Add files via upload

MarshMonci committed:
Add files via upload

MarshMonci committed:
Add files via upload

Removed double consts from UserRoleRelation. Uploading project files again since they did not reflect the addition of SystemManager to the source files.

MarshMonci commented:
Fixed the solution file. Everything current _should_ be included and working properly.
